go-controller/pkg/ovn/egressip_test.go

```diff
@@ -3410,6 +3410,167 @@ var _ = ginkgo.Describe("OVN master EgressIP Operations cluster default network"
+						getEgressIPLRPReRouteDbIDs(eIP.Name, egressPod.Namespace, egressPod.Name, IPFamilyValueV4,
+							types.DefaultNetworkName, fakeOvn.controller.eIPC.controllerName).GetExternalIDs()),
+					getEIPSNAT(podV4IP, egressPod.Namespace, egressPod.Name, egressIP.String(), "k8s-node1", DefaultNetworkControllerName),
+					&nbdb.LogicalRouter{
+						Name:     types.OVNClusterRouter,
+						UUID:     types.OVNClusterRouter + "-UUID",
+						Policies: []string{"reroute-UUID"},
+					},
+					&nbdb.LogicalRouterPort{
+						UUID:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name + "-UUID",
+						Name:     types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name,
+						Networks: []string{nodeLogicalRouterIfAddrV6, nodeLogicalRouterIfAddrV4},
+					},
+					&nbdb.LogicalRouter{
+						Name:    types.GWRouterPrefix + node1Name,
+						UUID:    types.GWRouterPrefix + node1Name + "-UUID",
+						Ports:   []string{types.GWRouterToJoinSwitchPrefix + types.GWRouterPrefix + node1Name + "-UUID"},
+						Nat:     []string{"egressip-nat-UUID"},
+						Options: map[string]string{"dynamic_neigh_routers": "false"},
+					},
+					&nbdb.LogicalSwitchPort{
+						UUID: "k8s-" + node.Name + "-UUID",
+						Name: "k8s-" + node.Name,
+						Addresses: []string{"fe:1a:b2:3f:0e:fb " + util.GetNodeManagementIfAddr(nodeSubnetV4).IP.String(),
+							"fe:1a:b2:3f:0e:fb " + util.GetNodeManagementIfAddr(nodeSubnetV6).IP.String()},
+					},
+					&nbdb.LogicalSwitch{
+						UUID:  node.Name + "-UUID",
+						Name:  node.Name,
+						Ports: []string{"k8s-" + node.Name + "-UUID"},
+					},
+				}
+				gomega.Eventually(fakeOvn.nbClient).Should(libovsdbtest.HaveData(expectedDatabaseState))
+				// create no-op updates that trigger reconcile for the selected pod async but update should continue to select the pod and not alter pod config
+				go func() {
+					ns, err := fakeOvn.fakeClient.KubeClient.CoreV1().Namespaces().Get(context.TODO(), egressNamespace.Name, metav1.GetOptions{})
+					if err != nil {
+						panic(err.Error())
+					}
+					// add new namespace label. Does not affect pod selection for EIP
+					ns = ns.DeepCopy()
+					ns.Labels["newlabel"] = "noop"
+					_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Namespaces().Update(context.TODO(), ns, metav1.UpdateOptions{})
+					if err != nil {
+						panic(err.Error())
+					}
+					// add new pod label. Does not affect pod selection for EIP
+					pod, err := fakeOvn.fakeClient.KubeClient.CoreV1().Pods(egressNamespace.Name).Get(context.TODO(), egressPod.Name, metav1.GetOptions{})
+					if err != nil {
+						panic(err.Error())
+					}
+					pod = pod.DeepCopy()
+					pod.Labels["newlabel"] = "noop"
+					_, err = fakeOvn.fakeClient.KubeClient.CoreV1().Pods(egressNamespace.Name).Update(context.TODO(), pod, metav1.UpdateOptions{})
+					if err != nil {
+						panic(err.Error())
+					}
+				}()
+				// ensure the DBs are unaltered
```

Comment:
before this change `AreNetworksCompatible` would return true? and we were removing pod config and it would flap?
like i don't understand how the pod update event is triggering a "changed network" event which is what we really want right?

Reply:
> before this change AreNetworksCompatible would return true? and we were removing pod config and it would flap?

its not - i just want to make sure that during the reconciliation look, theres no reconfig. I didnt want to write a unit test as specific as what you stated.

Reply:
aaah I see..

Reply:
> aaah I see..I'd actually rather have a test that actually tests the modularity of what is expected if the if condition matches - not the reverse.. though I understand pod config not being deleted is a way for checking this to ensure no flaps happen - I don't see why we need a new unit test for this then? cause I am sure at least one of the existing tests must have a "pod update" path - there are existing tests with label selectors that can be reused to also test this bit right?

my bug is regarding pod config flapping so i tesed to ensure that doesn't occur.
I couldnt think of scenario of network changing anyway. I didnt add that code. I meant to ask jaime but he was on pto.